web/src/pages/api/public/v2/prompts.ts

```diff
@@ -0,0 +1,76 @@
+import { verifyAuthHeaderAndReturnScope } from "@/src/features/public-api/server/apiAuth";
+import { cors, runMiddleware } from "@/src/features/public-api/server/cors";
+import { isPrismaException } from "@/src/utils/exceptions";
+import { prisma } from "@langfuse/shared/src/db";
+import { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  await runMiddleware(req, res, cors);
+  if (req.method === "GET") {
+    const authCheck = await verifyAuthHeaderAndReturnScope(
+      req.headers.authorization,
+    );
+    if (!authCheck.validKey)
+      return res.status(401).json({
+        message: authCheck.error,
+      });
+
+    console.log(
+      "trying to get prompt list",
+      authCheck.scope.projectId,
+      ", body:",
+      JSON.stringify(req.query, null, 2),
+    );
+
+    if (authCheck.scope.accessLevel !== "all") {
+      return res.status(401).json({
+        message:
+          "Access denied - need to use basic auth with secret key to GET prompts",
+      });
+    }
+    try {
+      const prompt = await prisma.prompt.findMany({
```

Comment:
I'd suggest to return a unique list of prompt names here including metadata similar to the prompt table in the UI.

Context: Prompt mgmt will change (multiple environments instead of a single `isActive`) and by using isActive here to directly return the prompts we'll likely have to change this api soon again.